app/src/main/AndroidManifest.xml

```diff
@@ -109,15 +121,6 @@
             android:windowSoftInputMode="adjustResize|stateAlwaysHidden" />
 
         <activity android:name=".IntentReceiverActivity" android:theme="@style/Theme.Transparent">
-            <intent-filter>
-                <action android:name="android.intent.action.VIEW" />
-
-                <category android:name="android.intent.category.DEFAULT" />
-                <category android:name="android.intent.category.BROWSABLE" />
-                <category android:name="mozilla.components.pwa.category.SHORTCUT" />
-                <data android:scheme="http" />
-                <data android:scheme="https" />
-            </intent-filter>
```

Comment:
I am not fully understanding what the fix is. **Not** having this filter here is the fix?

Reply:
The fix would be to keep the `action.VIEW` `category.DEFAULT` `http\https` intent filter for `org.mozilla.gecko.BrowserApp` since this is the preferred activity set for the default browser on Fennec.
Seems like on Huawei the default app setting is tied to the activity, not to just the package name.
I removed the same intent-filter from IntentReceiverActivity since when testing I saw that keeping it there also (after initially duplicating it for ..BrowserApp) means it having a higher precedence? (same intent filter for two activities..?) and Android would see it as the one who should handle the http schemes -> this would mean again a change from BrowserApp so the default app status would again be lost.